app/util/errors.ts

```diff
@@ -1,30 +1,156 @@
-export type ErrorCode = "Unknown" | "NotFound" | "AlreadyExists" | "PermissionDenied" | "Unauthenticated";
+import { google as google_status } from "../../proto/grpc_status_ts_proto";
+import { google as google_code } from "../../proto/grpc_code_ts_proto";
 
+export type ErrorCode = Omit<GRPCCodeName, "OK">;
+
+/**
+ * High-level error class representing any error from the BuildBuddy server,
+ * either GRPC or HTTP.
```

Comment:
The class hierarchy is a little weird here based on the documentation--do you actually intend to flip the code on its head at some point (HTTPStatusError extends BBError) or more cleanly compose things so that HTTPStatusError/FetchError etc are held by BBError?

Reply:
Yeah, thanks for pointing this out - I think what I would like to do is:
- Getting rid of `BuildBuddyError` entirely - it's sort of poorly defined
- In the places where we're doing stuff like `catch (e) { error_service.handleError(BuildBuddyError.parse(e)); }` we would just pass `e` instead of trying to parse the error. It's kind of ugly that we "parse" errors at all and I would rather have the errors be nicely structured at a lower level to begin with.
- The error_service could then do nicer stuff depending on the type of error that it's rendering, rather than receiving a generic `BuildBuddyError` which is more "watered down". e.g. for `FetchError` we could show a nice "network broken" icon or something like that (if we really wanted to, haha)

Reply:
Sure, that works, how about plopping something in the comment here about what the plan is?